### ISSUE TITLE
ManagedIdentityObjectId property of CertificatelessOptions should be renamed ManagedIdentityClientId

### ISSUE THREAD

**Issue by cbrooksmsft** [AzureAD/microsoft-identity-web#1991]
### Microsoft.Identity.Web Library

Microsoft.Identity.Web.CertificateLess

### Microsoft.Identity.Web version

1.25.10

### Web app

Sign-in users and call web APIs

### Web API

Protected web APIs call downstream web APIs

### Token cache serialization

Not Applicable

### Description

The ManagedIdentityObjectId property in [CertificatelessOptions](https://github.com/AzureAD/microsoft-identity-web/blob/c0d8346a11c762d5418d379d6914cefeb0d2fa5c/src/Microsoft.Identity.Web.Certificateless/CertificatelessOptions.cs#L26) should actually be named "ManagedIdentityClientId".

It is passed in the [TokenAquisition](https://github.com/AzureAD/microsoft-identity-web/blob/fdc81a924cf6e893102c33a6eebd3b7ebe5dbb41/src/Microsoft.Identity.Web/TokenAcquisition.cs#L758) class to the managedIdentityClientId constructor parameter for [ManagedIdentityClientAssertion](https://github.com/AzureAD/microsoft-identity-web/blob/88e6a4ea58df80416b7808e7714e3001670f1626/src/Microsoft.Identity.Web.Certificateless/ManagedIdentityClientAssertion.cs).

`builder.WithClientAssertion(new ManagedIdentityClientAssertion(mergedOptions.ClientCredentialsUsingManagedIdentity.ManagedIdentityObjectId).GetSignedAssertion);`

This value is then in turn passed to the ManagedIdentityClientId property of the [DefaultAzureCredentialOptions](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/identity/Azure.Identity/src/Credentials/DefaultAzureCredentialOptions.cs):

`var credential = new DefaultAzureCredential(new DefaultAzureCredentialOptions { ManagedIdentityClientId = _managedIdentityClientId });
`

 This option required the use of the clientId for the managed identity, and not the objectId.

### Reproduction steps

1. Configure Certless app authentication with Managed Identity using CertificateLessOptions
2. Provide the objectId of the managed identity to the AzureAd -> ClientCredentialsUsingManagedIdentity -> ManagedIdentityObjectIdentity config property
3. Attempt to request a token

### Error message

compute platform returns error indicating that the requested managed identity is not available,

[Error]   JWT Token exception: Azure.Identity.AuthenticationFailedException: ManagedIdentityCredential authentication failed: Service request failed.
Status: 400 (Bad Request)

Content:
{"statusCode":400,"message":"Unable to load the proper Managed Identity.","correlationId":"3f54e937-2426-4ac6-8b3b-e11f7ff28c02"}

### Relevant code snippets

```csharp
{
  "AzureAd": { 
    "Instance": "https://login.windows-ppe.net/", 
    "ClientId": "[Client_id-of-web-app-eg-2ec40e65-ba09-4853-bcde-bcb60029e596]", 
    "TenantId": "common", 
   // To call an API 
    "ClientCredentialsUsingManagedIdentity": { 
      "IsEnabled": true, 
      "ManagedIdentityObjectIdentity": "2387ff05-c838-43cd-b072-63de4be15119" 
    },
  }, 
  "GraphBeta": { 
    "BaseUrl": "https://graph.microsoft-ppe.com/beta", 
    "Scopes": "user.read" 
  } 
}
```

### Expected behavior

Developer would expect the system to request tokens using the provided ObjectID (as is available at the MSI Rest layer [here](https://learn.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/how-to-use-vm-token#get-a-token-using-http))

As the DefaultAzureCredential dependency here does not support requesting tokens by objectId, but only by ClientId, the developer should be asked to provide the clientId instead.

Renaming this parameter with a breaking change in the next major version of the library would be appropriate.

Documentation updates pointing out the need to pass a clientId to this parameter for the current version would help prevent developer confusion.

**Comment by jmprieur** [comment]
Thansk @cbrooksmsft 

Do you mean that the ClientId is the underlying application 
Where do the developers get the ClientID from, for instance when they host their app in app services (App Services provides the Object ID).

In general how do people get the ClientID?

![image](https://user-images.githubusercontent.com/13203188/205515194-02d9ede8-8d86-4b9e-b788-6aa0ab5ed562.png)

**Comment by cbrooksmsft** [comment]
The clientId is used when referring to a specific associated User-Assigned Managed Identity.  When referring to the System-Assigned identity on a resource (as shown in the image above), the default constructor (with no clientId) is used.

The clientId of a user-assigned managed identity is found on the summary page for the identity:

![image](https://user-images.githubusercontent.com/9362227/205517462-efeb7daa-02d0-4ddb-acf9-46a754b447f2.png)

**Comment by jmprieur** [comment]
Thanks for the clarification, @cbrooksmsft. I should have found it by myself.
We'll change the name of the parameters in 2.x

**Comment by jmprieur** [comment]
Hi Chris
This is done and should be available internally (from the Azure DevOps IDDP artifact feed) later today part of the first released version of Microsoft.Identity.Web 2.x


Regards

Jean-Marc



________________________________

**Comment by jennyf19** [comment]
Released in [2.5.0](https://github.com/AzureAD/microsoft-identity-web/releases/tag/2.5.0)
